core/src/main/scala/org/apache/spark/rdd/ReliableCheckpointRDD.scala

```diff
@@ -83,8 +84,9 @@ private[spark] class ReliableCheckpointRDD[T: ClassTag](
    * Return the locations of the checkpoint file associated with the given partition.
    */
   protected override def getPreferredLocations(split: Partition): Seq[String] = {
-    val status = fs.getFileStatus(
-      new Path(checkpointPath, ReliableCheckpointRDD.checkpointFileName(split.index)))
+    val path = new Path(checkpointPath, ReliableCheckpointRDD.checkpointFileName(split.index))
```

Comment:
I don't think it's really supported; if this much can be more flexible without adding much of any complexity though, all the better.